pkg/gpu/context.go

```diff
@@ -178,7 +180,7 @@ func (ctx *systemContext) filterDevicesForContainer(devices []*ddnvml.Device, co
 	numContainerGPUs := 0
 	for _, resource := range container.AllocatedResources {
 		// Only consider NVIDIA GPUs
-		if resource.Name != nvidiaResourceName {
+		if resource.Name != nvidiaResourceName && !strings.HasPrefix(resource.Name, nvidiaMigResourceName) {
```

Comment:
_nit:_ consider introducing a helper func (as you are using similar conditional logic here and in gpu.go)